### PR TITLE
#47 Log improvements

### PR DESCRIPTION
**Pull request by maelstrom0x8**
This PR adds logging statements to the controller methods in order to capture HTTP details.

### CODE REVIEW

File: techbuzz/src/main/java/com/sivalabs/techbuzz/posts/web/controllers/AddVoteController.java [hunk 1]
```diff
@@ -27,6 +30,10 @@ class AddVoteController {
     public String createVote(@Valid @RequestBody CreateVoteRequest request, @CurrentUser User loginUser, Model model) {
         var createVoteRequest = new CreateVoteRequest(request.postId(), loginUser.getId(), request.value());
         postService.addVote(createVoteRequest);
+        log.info(
+                "POST /partials/add-vote - Vote added by User id: {} for Post id: {}",
```
Comment (sivaprasadreddy) [inline techbuzz/src/main/java/com/sivalabs/techbuzz/posts/web/controllers/AddVoteController.java:34]: No need to include HTTP Method and URL pattern in logs.

File: techbuzz/src/main/java/com/sivalabs/techbuzz/posts/web/controllers/ViewCategoryController.java [hunk 2]
```diff
@@ -32,9 +32,11 @@ public String viewCategory(
             @PathVariable(name = "categorySlug") String categorySlug,
             @RequestParam(name = "page", defaultValue = "1") Integer page,
             Model model) {
-        log.info("Fetching posts for category {} with page: {}", categorySlug, page);
+        log.info("GET /c/{} - Fetching posts for category {} with page: {}", categorySlug, categorySlug, page);
         PagedResult<PostViewDTO> data = postService.getPostsByCategorySlug(categorySlug, page);
         if (data.data().isEmpty() && (page > 1 && page > data.totalPages())) {
+            log.warn("No posts found for category: {}", categorySlug);
```
Comment (sivaprasadreddy) [inline techbuzz/src/main/java/com/sivalabs/techbuzz/posts/web/controllers/ViewCategoryController.java:38]: These 2 log statements can be combined into a single one:

log.warn("No posts found for category: {}, page:{}. Redirecting to last page", categorySlug, page);

File: techbuzz/src/main/java/com/sivalabs/techbuzz/users/web/controllers/EmailVerificationController.java [hunk 3]
```diff
@@ -19,11 +24,14 @@ public EmailVerificationController(final UserService userService) {
 
     @GetMapping("/verify-email")
     public String verifyEmail(Model model, @RequestParam("email") String email, @RequestParam("token") String token) {
+        logger.info("GET /verify-email - Verifying email {} with token: {}", email, token);
```
Comment (sivaprasadreddy) [inline techbuzz/src/main/java/com/sivalabs/techbuzz/users/web/controllers/EmailVerificationController.java:27]: "token" is a sensitive information, should not be logged.

File: techbuzz/src/main/java/com/sivalabs/techbuzz/users/web/controllers/ResendVerificationController.java [hunk 4]
```diff
@@ -59,6 +59,7 @@ public String resendVerification(
                         "errorMessage", "Account is already verified, please use forget password if needed");
             } else {
                 this.sendVerificationEmail(request, user);
+                logger.info("Sent email verification link");
```
Comment (sivaprasadreddy) [inline techbuzz/src/main/java/com/sivalabs/techbuzz/users/web/controllers/ResendVerificationController.java:62]: Update to logger.info("Sent email verification link to {}", email);

**Review by sivaprasadreddy — CHANGES_REQUESTED** [review]
Thanks for the PR.
Could you please look into the review comments?

**Comment by maelstrom0x8** [comment]
Alright sir. I'm on it.

**Comment by maelstrom0x8** [comment]
hello @sivaprasadreddy 
please review the updated changes. thank you!

**Comment by sivaprasadreddy** [comment]
@maelstrom0x8 The changes looks good.
However, there are code formatting issue.
Please run "./mvnw spotless:apply" which will auto format the code and commit the changes.

**Comment by maelstrom0x8** [comment]
done.

**Pull request merged**